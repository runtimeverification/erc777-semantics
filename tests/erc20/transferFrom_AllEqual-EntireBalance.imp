test.preamble();

// Testing transferFrom(From,To,Value) when From = To = caller.
// Transfer entire balance to yourself.

test.approve(caller, 70);   // allow entire balance
test.transferFrom(caller, caller, 70);  // transfer entire balance
