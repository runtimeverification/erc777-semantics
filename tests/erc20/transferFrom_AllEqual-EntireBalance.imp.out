7 is approving allowance 70 for 7
7 is transferring 70 from 7 to 7
Balance of 7 is 70
Balance of 7 is 70
Allowance of 7 for 7 is 0
